src/asset.ts

```diff
@@ -0,0 +1,152 @@
+        ', ',
+      )} cannot be opted in. Ensure that they are valid and that the account has not previously opted into them.`,
+    )
+  }
+}
+
+async function ensureAssetBalance(account: Account, assetIds: number[], client: Algodv2) {
+  const assetPromises = assetIds.map(async (assetId) => {
+    try {
+      const accountAssetInfo = await client.accountAssetInformation(account.addr, assetId).do()
+      if (accountAssetInfo['asset-holding']['amount'] !== 0) {
+        Config.logger.debug(`asset ${assetId} is not with zero balance`)
+        return assetId
+      }
+    } catch (e) {
+      Config.logger.debug(`Account ${account.addr} does not have asset ${assetId}`)
+      return assetId
+    }
+    return null
+  })
+
+  const invalidAssets = (await Promise.all(assetPromises)).filter((assetId) => assetId !== null)
+  if (invalidAssets.length > 0) {
+    throw new Error(
+      `Assets ${invalidAssets.join(
+        ', ',
+      )} cannot be opted out. Ensure that they are valid and that the account has previously opted into them.`,
+    )
+  }
+}
+
+export async function optIn(client: Algodv2, optInAccount: Account, assetIds: number[]) {
+  const result: Record<number, string> = {}
+  await ensureAccountIsValid(client, optInAccount)
+  await ensureAssetFirstOptIn(client, optInAccount, assetIds)
+
+  const assets = await Promise.all(assetIds.map((aid) => client.getAssetByID(aid).do()))
+  const suggestedParams = await client.getTransactionParams().do()
+
+  for (const assetGroup of chunks(assets, MaxTxGroupSize)) {
+    try {
+      const transactionToSign: TransactionToSign[] = assetGroup.flatMap((asset) => [
+        {
+          transaction: algosdk.makeAssetTransferTxnWithSuggestedParamsFromObject({
+            from: optInAccount.addr,
+            to: optInAccount.addr,
+            assetIndex: asset.index,
+            amount: 0,
+            rekeyTo: undefined,
+            revocationTarget: undefined,
+            closeRemainderTo: undefined,
+            suggestedParams,
+          }),
+          signer: optInAccount,
+        },
+      ])
+      const txnGrp: TransactionGroupToSend = {
+        transactions: transactionToSign,
+        signer: optInAccount,
+      }
+      const sendGroupOfTransactionsResult = await sendGroupOfTransactions(txnGrp, client)
+      assetGroup.map((asset, index) => {
+        result[asset.index] = sendGroupOfTransactionsResult.txIds[index]
+
+        Config.logger.info(
+          `Successfully opted in of asset ${asset.index} with transaction ID ${sendGroupOfTransactionsResult.txIds[index]},
+          grouped under ${sendGroupOfTransactionsResult.groupId} round ${sendGroupOfTransactionsResult.confirmations}.`,
+        )
+      })
+    } catch (e) {
+      throw new Error(`Received error trying to opt in ${e}`)
+    }
+  }
+  return result
+}
+
+export async function optOut(client: Algodv2, optoutAccount: Account, assetIds: number[]): Promise<Record<number, string>> {
```

Comment:
in py version you named the account field as `account`, lets either rename it to account here or rename it to match the current title here on py version for consistency